coingecko/coingecko-fetcher.ts

```diff
@@ -1,123 +0,0 @@
-import { kv } from "@vercel/kv";
-import moment from "moment";
-import { userKey } from "./constants";
-import { coingeckoAPIErrorResponse, invalidResponseTypesError } from "./errors";
-import { PriceDataResponse } from "./interfaces";
-import { coinList } from "./supported-coins";
-import {
-  CoingeckoResponse,
-  getCoingeckoRangeURL,
-  getDayId,
-  getTimestampFromDayId,
-  KVDataToPrice,
-  parsePriceResponse,
-} from "./utils";
-
-const fetchData = async (symbol: string, start: number, finish: number) => {
-  const id = coinList[symbol];
-  const url = getCoingeckoRangeURL(id, start, finish);
-  const res = await fetch(
-    `https://api.scraperapi.com/?api_key=${
-      process.env.SCRAPER_API_KEY
-    }&url=${encodeURIComponent(url)}`
-  );
-
-  if (res.status !== 200) {
-    throw coingeckoAPIErrorResponse(res);
-  }
-  const rawResponse = await res.json();
-
-  const parsed = CoingeckoResponse.safeParse(rawResponse);
-
-  if (parsed.success === false) {
-    console.error(parsed.error.issues);
-    throw invalidResponseTypesError;
-  }
-
-  return parsePriceResponse(parsed.data.prices);
```

Comment:
start by moving 
```ts
const parsed = CoingeckoResponse.safeParse(rawResponse);

if (parsed.success === false) {
  console.error(parsed.error.issues);
  throw invalidResponseTypesError;
}
```

inside a `parsePriceResponse` function, than you will notice that second step of validation can be a simple z.transform